src/lib/utils/package-manifest.ts

```diff
@@ -190,12 +192,23 @@ export function getTsEntryPointForPackage(
     ) {
         packageMain = packageJson.main;
     }
-    const jsEntryPointPath = resolve(packageJsonPath, "..", packageMain);
-    if (!isFile(jsEntryPointPath)) {
-        logger.warn(
-            `Could not determine the JS entry point for ${packageJsonPath}. Package will be ignored.`
-        );
-        return ignorePackage;
+    let jsEntryPointPath = resolve(packageJsonPath, "..", packageMain);
+    // The jsEntryPointPath from the package manifest can be like a require path.
+    // It could end with .js, or it could end without .js, or it could be a folder containing an index.js
+    // We can use require.resolve to let node do its magic.
+    // Pass an empty `paths` as node_modules locations do not need to be examined
+    try {
+        jsEntryPointPath = require.resolve(jsEntryPointPath, { paths: [] });
```

Comment:
This is the other change in here. package.json's `main` values seem to be roughly as flexible as the `require()` paths themselves in terms of what actual file gets loaded.